src/HealthChecks.OpenIdConnectServer/IdSvrHealthCheck.cs

```diff
@@ -27,13 +28,35 @@ public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context
             var httpClient = _httpClientFactory();
             using var response = await httpClient.GetAsync(_discoverConfigurationSegment, HttpCompletionOption.ResponseHeadersRead, cancellationToken).ConfigureAwait(false);
 
-            return response.IsSuccessStatusCode
-                ? HealthCheckResult.Healthy()
-                : new HealthCheckResult(context.Registration.FailureStatus, description: $"Discover endpoint is not responding with 200 OK, the current status is {response.StatusCode} and the content {await response.Content.ReadAsStringAsync().ConfigureAwait(false)}");
+            if (!response.IsSuccessStatusCode)
+            {
+                return new HealthCheckResult(context.Registration.FailureStatus, description: $"Discover endpoint is not responding with 200 OK, the current status is {response.StatusCode} and the content {await response.Content.ReadAsStringAsync().ConfigureAwait(false)}");
+            }
+
+            var discoveryResponse = await response
+                   .Content
+                   .ReadFromJsonAsync<DiscoveryEndpointResponse>()
+                   .ConfigureAwait(false)
+               ?? throw new ArgumentException("Could not deserialize to discover endpoint response!");
+
+            ValidateResponse(discoveryResponse);
+
+            return HealthCheckResult.Healthy();
         }
         catch (Exception ex)
         {
             return new HealthCheckResult(context.Registration.FailureStatus, exception: ex);
         }
     }
+
+    private static void ValidateResponse(DiscoveryEndpointResponse response)
```

Comment:
Let's make it an instance method of `DiscoveryEndpointResponse` + place https://openid.net/specs/openid-connect-discovery-1_0.html#IssuerDiscovery link in comments for that method.

Reply:
Thanks, done.